indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/WorkerSelectUtils.java

```diff
@@ -95,6 +95,79 @@ public static ImmutableWorkerInfo selectWorker(
     }
   }
 
+  /**
+   * Helper for {@link WorkerSelectStrategy} implementations.
+   *
+   * @param allWorkers     map of all workers, in the style provided to {@link WorkerSelectStrategy}
+   * @param workerTierSpec worker tier spec, or null
+   * @param workerSelector function that receives a list of eligible workers: version is high enough, worker can run
+   *                       the task, and worker satisfies the worker tier spec. may return null.
+   *
+   * @return selected worker from "allWorkers", or null.
+   */
+  @Nullable
+  public static ImmutableWorkerInfo selectWorker(
+      final Task task,
+      final Map<String, ImmutableWorkerInfo> allWorkers,
+      final WorkerTaskRunnerConfig workerTaskRunnerConfig,
+      @Nullable final WorkerTierSpec workerTierSpec,
+      final Function<ImmutableMap<String, ImmutableWorkerInfo>, ImmutableWorkerInfo> workerSelector
+  )
+  {
+    // Workers that could potentially run this task, ignoring workerTierSpec.
+    final Map<String, ImmutableWorkerInfo> runnableWorkers = allWorkers
+        .values()
+        .stream()
+        .filter(worker -> worker.canRunTask(task)
+                          && worker.isValidVersion(workerTaskRunnerConfig.getMinWorkerVersion()))
+        .collect(Collectors.toMap(w -> w.getWorker().getHost(), Function.identity()));
+
+    // select worker according to worker tier spec
+    if (workerTierSpec != null && workerTierSpec.getTierMap() != null) {
+      final WorkerTierSpec.TierConfig tierConfig = workerTierSpec.getTierMap()
+                                                                 .getOrDefault(
+                                                                     task.getType(),
+                                                                     new WorkerTierSpec.TierConfig(null, null)
+                                                                 );
+      final String defaultTier = tierConfig.getDefaultTier();
+      final Map<String, String> tierAffinity = tierConfig.getTierAffinity();
+
+      // select worker from preferred tier
+      final String preferredTier = tierAffinity != null ? tierAffinity.get(task.getDataSource()) : defaultTier;
```

Comment:
maybe default the tierAffinity in TierConfig to be Collections.EMPTY_MAP so that null check here is not needed.

Reply:
Fixed.